de.bund.bfr.knime.fsklab.nodes.tests/src/de/bund/bfr/knime/fsklab/v1_9/FskPortObjectUtilTest.java

```diff
@@ -0,0 +1,49 @@
+package de.bund.bfr.knime.fsklab.v1_9;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.bund.bfr.knime.fsklab.nodes.FskPortObjectUtil;
+import de.bund.bfr.knime.fsklab.nodes.environment.ArchivedEnvironmentManager;
+import de.bund.bfr.knime.fsklab.nodes.environment.EnvironmentManager;
+
+public class FskPortObjectUtilTest {
+
+	@Test
+	public void testDeserialize_CorrectClass_shouldReturnEqual() throws Exception {
+		ObjectMapper mapper = new ObjectMapper();
+		ClassLoader classLoader = getClass().getClassLoader();
+		
+		try (InputStream inputStream = new FileInputStream(new File("files/example_environment.json"))){
+			EnvironmentManager manager = FskPortObjectUtil.deserializeAfterClassloaderReset(classLoader, mapper, inputStream, EnvironmentManager.class);
+			assertEquals(ArchivedEnvironmentManager.class, manager.getClass());

```

Comment:
We could instead instanceof to check if the manager is an object of ArchivedEnvironmentManager:
```java
assertTrue(manager instanceof ArchivedEnvironmentManager)
```